internal/services/ch/ch.go

```diff
@@ -0,0 +1,121 @@
+package ch
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/DIMO-Network/clickhouse-infra/pkg/connect"
+	"github.com/DIMO-Network/rewards-api/internal/config"
+	"github.com/volatiletech/sqlboiler/v4/drivers"
+	"github.com/volatiletech/sqlboiler/v4/queries"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+)
+
+// Client is a ClickHouse client that interacts with the ClickHouse database.
+type Client struct {
+	conn clickhouse.Conn
+}
+
+const (
+	integPrefix = "dimo/integration/"
+)
+
+var dialect = drivers.Dialect{
+	LQ: '`',
+	RQ: '`',
+}
+
+// NewClient creates a new ClickHouse client.
+func NewClient(settings *config.Settings) (*Client, error) {
+	conn, err := connect.GetClickhouseConn(&settings.Clickhouse)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create ClickHouse connection: %w", err)
+	}
+
+	err = conn.Ping(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("failed to ping clickhouse: %w", err)
+	}
+	return &Client{conn: conn}, nil
+}
+
+// DescribeActiveDevices returns list of vehicles and associated vehicle integrations that have signals in Clickhouse during specified time period
+func (s *Client) DescribeActiveDevices(ctx context.Context, start, end time.Time) ([]*Vehicle, error) {
+	q := &queries.Query{}
+	queries.SetDialect(q, &dialect)
+	qm.Apply(q, []qm.QueryMod{
+		qm.Select("token_id", "groupUniqArray(source)"),
+		qm.From("signal"),
+		qm.Where("timestamp >= ?", start),
+		qm.And("timestamp < ?", end),
+		qm.GroupBy("token_id"),
+	}...)
```

Comment:
We make this slice and then immediately expand it. What about
```suggestion
	qm.Apply(q,
		qm.Select("token_id", "groupUniqArray(source)"),
		qm.From("signal"),
		qm.Where("timestamp >= ?", start),
		qm.And("timestamp < ?", end),
		qm.GroupBy("token_id"),
	)
```